File: src/Models/Consumer.java
package Models;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class Consumer implements Serializable {

    private static final long serialVersionUID = 578515438738407941L;
    private String ip;
    private int port;
    private List<Integer> interests = new ArrayList<>();

    public Consumer(String ip, Integer port) {
        this.ip = ip;
        this.port = port;
    }

    public String getIP() {
        return ip;
    }

    public int getPort() {
        return port;
    }

    public List<Integer> getInterests() {
        return this.interests;
    }

    public void addInterest(Integer topic) {
        this.interests.add(topic);
    }

    @Override
    public String toString() {
        return "Publisher " + this.getIP() + ":" + this.getPort();
    }

}

File: src/models/Consumer.kt
package models

import java.io.Serializable
import java.util.ArrayList

class Consumer(val ip: String, val port: Int?) : Serializable {

    val interests: MutableList<Int> = ArrayList()

    fun addInterest(topic: Int) {
        this.interests.add(topic)
    }

    override fun toString(): String {
        return "Consumer ${this.ip}:${this.port}"
    }

    companion object {
        private const val serialVersionUID = 578515438738407941L
    }

}
